drivers/regex/mlx5/mlx5_rxp.h

```diff
@@ -0,0 +1,133 @@
+				      MLX5_RXP_CTRL_JOB_DESC_HPM_ENABLE | \
+				      MLX5_RXP_CTRL_JOB_DESC_ANYMATCH_ENABLE)
+
+#define MLX5_RXP_CTRL_VALID 0x8000
+
+#define MLX5_RXP_RESP_STATUS_MAX_PRI_THREADS (1 << 3)
+#define MLX5_RXP_RESP_STATUS_MAX_SEC_THREADS (1 << 4)
+#define MLX5_RXP_RESP_STATUS_MAX_LATENCY (1 << 5)
+#define MLX5_RXP_RESP_STATUS_MAX_MATCH (1 << 6)
+#define MLX5_RXP_RESP_STATUS_MAX_PREFIX	(1 << 7)
+#define MLX5_RXP_RESP_STATUS_HPM (1 << 8)
+#define MLX5_RXP_RESP_STATUS_ANYMATCH (1 << 9)
+#define MLX5_RXP_RESP_STATUS_PMI_SOJ (1 << 13)
+#define MLX5_RXP_RESP_STATUS_PMI_EOJ (1 << 14)
+
+#define MLX5_RXP_SUBSET_ID_MAX 65535
+
+/*
+ * This describes the header the RXP expects for any search data.
+ */
+struct mlx5_rxp_job_desc {
+	uint32_t job_id;
+	uint16_t ctrl;
+	uint16_t len;
+	uint16_t subset[4];
+}  __attribute__ ((__packed__));
+
+struct mlx5_rxp_response_desc {
+	uint32_t job_id;
+	uint16_t status;
+	uint8_t	detected_match_count;
+	uint8_t	match_count;
+	uint16_t primary_thread_count;
+	uint16_t instruction_count;
+	uint16_t latency_count;
+	uint16_t pmi_min_byte_ptr;
+}  __attribute__ ((__packed__));
+
+//New 64bit match tuple for Mellanox
+struct mlx5_rxp_match_tuple {
+	uint32_t rule_id;
+	uint16_t start_ptr;
+	uint16_t length;
+}  __attribute__ ((__packed__));
+
+struct mlx5_rxp_response {
+	struct mlx5_rxp_response_desc header;
+	struct mlx5_rxp_match_tuple matches[0];
+};
+
+#define MLX5_RXP_MAX_MATCHES 254
+
+#define MLX5_RXP_CTL_RULES_PGM	1
+#define MLX5_RXP_CTL_RULES_PGM_INCR	2
+
+#define MLX5_RXP_ROF_ENTRY_INST            0
+#define MLX5_RXP_ROF_ENTRY_EQ              1
+#define MLX5_RXP_ROF_ENTRY_GTE             2
+#define MLX5_RXP_ROF_ENTRY_LTE             3
+#define MLX5_RXP_ROF_ENTRY_CHECKSUM        4
+#define MLX5_RXP_ROF_ENTRY_CHECKSUM_EX_EM  5
+#define MLX5_RXP_ROF_ENTRY_IM              6
+#define MLX5_RXP_ROF_ENTRY_EM              7
+#define MLX5_RXP_ROF_ENTRY_TYPE_MAX        7
+
+struct mlx5_rxp_ctl_hdr {
+	uint16_t	cmd;
+	uint32_t	len;
+};
+
+struct mlx5_rxp_rof_entry {
+	uint8_t		type;
+	uint32_t	addr;
+	uint64_t	value;
+};
+
+struct mlx5_rxp_rof {
+	uint32_t rof_version;
+	char *timestamp;
+	char *rxp_compiler_version;
+	uint32_t rof_revision;
+	uint32_t number_of_entries;
+	struct mlx5_rxp_rof_entry *rof_entries;
+};
+
+struct mlx5_rxp_ctl_rules_pgm {
+	struct mlx5_rxp_ctl_hdr	hdr;
+	uint32_t		count;
+	struct mlx5_rxp_rof_entry	rules[0];
+} __attribute__ ((__packed__));
+
+/*
+ * RXP programming mode setting
+ */
+enum mlx5_rxp_program_mode {
+    MLX5_RXP_MODE_NOT_DEFINED = 0,
+    MLX5_RXP_SHARED_PROG_MODE,
+    MLX5_RXP_PRIVATE_PROG_MODE,
+};
+
+#define MLX5_RXP_POLL_CSR_FOR_VALUE_TIMEOUT 3000 /* Poll timeout in ms. */
+#define MLX5_RXP_INITIALIZATION_TIMEOUT 60000 /* Initialization timeout in ms. */
+#define MLX5_RXP_MAX_ENGINES 2u /* Number of engines. */
+#define MLX5_RXP_SHADOW_EM_COUNT 1u /* Extra External Memories to use. */
+#define MLX5_RXP_MAX_NOT_USED 0xFF
+
+struct mlx5_regex_umem {
```

Comment:
we still need this?

Reply:
no, this should be removed in Francis patch